### ISSUE TITLE
Having an offical portable Linux build of ungoogled-chromium

### ISSUE THREAD

**Issue by harry963** [Eloston/ungoogled-chromium#1532]
Hello, I hope everyone is doing well during these hard times. This topic may have been mentioned before But I would really like an official portable build of ungoogled-chromium.
I cannot compile the browser due to low end hardware and I don't want to use some random binary that isn't official.
I think either just a regular portable build or an app image would be good maybe even add it to Guix or Nix since they work on every distro. This would allow for people with low end hardware or people who don't want to deal with the hassle of compiling to use ungoogled-chromium.
I also see many issues on GitHub of people asking to update certain binaries this would fix everyone asking for Linux binaries as they could just use this new portable binary. I think this build would benefit most ungoogled-chromium users who may be currently using a third party binary.
It would also allow for distros to add ungoogled-chromium to their repos since there is now an official binary especially smaller distros with smaller dev teams.
It wouldn't fix the issue for Windows, Mac or Linux with Musl (like Void musl or Alpine) users but it would certainly help many.
An official portable binary would be a very nice addition to have and I ask Eloston to please consider this in the future.

**Comment by mdedonno1337** [comment]
One of the issues is the compute power needed to do the compilation, it would take arround 15$ per compilation on a Anazon AWS compute.
On some other build services you have limitation of time, cpu and ram which are problematic to compile.

**Comment by harry963** [comment]
I could see how this could be an issue. Do you know how much it would cost per build? Maybe it would be possible for the community to fund it. 
Also, How are these binaries usually made? I would've assumed that most of these third party binaries are just complied off regular consumer hardware.
Also, it wouldn't have to be hosted on Guix. If a portable build was available, then distros could package it in their repo or people could just download and run the binary

**Comment by PF4Public** [comment]
I'm building [binaries of ungoogled-chromium](https://github.com/PF4Public/ungoogled-chromium-bin/releases) for Gentoo, but they are easily useable in virtually any Linux, which provides the same versions of libraries, that I used to build against in Gentoo.

>This topic may have been mentioned before
True and the conclusion was the difficulty to define "official"? Are my binaries official enough? I see some folks use them (based on the issues filled every now and then about them).

**Comment by harry963** [comment]
I think official would mean built by Eloston himself. I would very much like to use ungoogled-chromium however I am wary of the third party builds, and I feel many others feel the same way. I honestly think that ungoogled-chromium would have a much larger user base if it had a proper official build.

Also where are your builds located? I might see if I can get them on working on Artix and Arch but I think I'll stick to FF and Palemoon.

**Comment by PF4Public** [comment]
> Also where are your builds located?


> I'm building [binaries of ungoogled-chromium](https://github.com/PF4Public/ungoogled-chromium-bin/releases) for Gentoo

**Comment by networkException** [comment]
As already mentioned in https://github.com/Eloston/ungoogled-chromium/pull/1534#issuecomment-854975692 there are somewhat official builds. I'll just assume that those packages would meet the requirement "built by Eloston", but thats up to everyone personally of course.

Still this does not mean that everything is perfect, I personally think that submitting binaries without much verification is both really bad for "marketing" and security. Then there's the previously brought up problem of build times / resource usage. I can build a binary on my server in ~10h, same time for open build service and github actions takes 23h - 27h. I guess in a perfect world there would be a fast, community founded server attached to ci that all packaging repositories can use.

**Comment by mdedonno1337** [comment]
If is would have been on a gitlab server, I would had already proposed to ask people to join ci-worker (with a fleat of workers, at least one would be online at any time I prsume); I know (and I am OK to help) this, but I dont have any knowledge regarding the build process on Github.

**Comment by harry963** [comment]
That is all true. So considering all these things do you think it would be possible? I think even if we had GitHub or Git Lab compiling and just had to wait an extra day for a binary it would still be better then nothing. 
I think Git Lab would be better since it's FOSS but that I think GitHub is still better then nothing. 
Maybe Eloston could sign the binaries with a PGP key so we could verify them?

**Comment by miclill** [comment]
As mentioned by @networkException there are "official" binaries here (glibc and even musl):
https://github.com/Eloston/ungoogled-chromium-binaries/

**Comment by networkException** [comment]
Those releases are more of a temporary solution for issues with debian builds, see https://github.com/ungoogled-software/ungoogled-chromium-debian/issues/255
